webapp/README.md

```diff
@@ -33,48 +33,74 @@ $ yarn build
 
-![Storybook output](../.gitbook/assets/storybook-output.png)
+```bash
+# run unit tests
+$ docker-compose exec webapp yarn test
+```
 
-Click on the link http://localhost:3002/ to open the browser to your interactive storybook.
+```bash
+# start storybook
+$ docker-compose exec webapp yarn storybook
+```
+
+You can then visit the Storybook playground on `http://localhost:3002`
 
 {% endtab %}
 
 {% tab title="Without Docker" %}
-Run the following command:
+
+After starting the application following the above guidelines, open new terminal windows and navigate to the `/webapp` directory for each of these commands:
 
 ```bash
-# in webapp/
-yarn storybook
+# run eslint in /webapp
+$ yarn lint
 ```
 
-Open http://localhost:3002/ in your browser
+```bash
+# run unit tests in /webapp
+$ yarn test
+```
 
-{% endtab %}
-{% endtabs %}
+```bash
+# start storybook in /webapp
+$ yarn storybook
+```
 
+You can then visit the Storybook playground on `http://localhost:3002`
 
+{% endtab %}
+{% endtabs %}
 
-## Styleguide
+## Styleguide Migration
 
-All reusable Components \(for example avatar\) should be done inside the [Nitro-Styleguide](https://github.com/Human-Connection/Nitro-Styleguide) repository.
+We are currently in the process of migrating our styleguide components and design tokens from the [Nitro Styleguide](https://github.com/Human-Connection/Nitro-Styleguide) into the main [Human Connection repository](https://github.com/Human-Connection/Human-Connection) and refactoring our components in the process. During this migration, our new components will live in a `view` folder to separate them from the old, yet untouched components.
 
-![Styleguide Screenshot](../.gitbook/assets/screenshot-styleguide%20%281%29.png)
+### Folder Structure
 
-More information can be found here: [https://github.com/Human-Connection/Nitro-Styleguide](https://github.com/Human-Connection/Nitro-Styleguide)
+The folder structure we are aiming for is based on the [directory setup proposed by Nuxt.js](https://nuxtjs.org/guide/directory-structure):
 
-If you need to change something in the styleguide and want to see the effects on the frontend immediately, then we have you covered. You need to clone the styleguide to the parent directory `../Nitro-Styleguide` and run `yarn && yarn run dev`. After that you run `yarn run dev:styleguide` instead of `yarn run dev` and you will see your changes reflected inside the frontend!
+- **assets** contains icons, images and logos in `svg` format
+- **components** are the generic building blocks of the app – small, reusable and usually not coupled to state
+- **features** are composed of components but tied to a particular function of the app (e.g. `comment` or `post`)
+- **layouts** can use components to create layout templates for pages
+- **pages** are the entry points for all `routes` in the app and are composed of layouts, features and components
+- **styles** holds all shared SCSS files such as `variables` and `mixins`
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-31T15:38:32Z" title="Thursday, October 31st 2019, 4:38:32 pm +01:00">Oct 31, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -33,48 +33,74 @@ $ yarn build
 $ yarn start
 ```
 
-### Storybook
+### Run tests
 
-We encourage contributors to use Storybook to test out new components in an isolated way, and benefit from its many features.
-See the docs for live examples and answers to FAQ, among other helpful information. ![Storybook docs](https://storybook.js.org/docs/basics/introduction/)
+We ensure the quality of our frontend code by using
+- [ESLint](https://eslint.org/) for checking our JavaScript code
+- [Jest](https://jestjs.io/) and [Vue Test Utils](https://vue-test-utils.vuejs.org/) to unit test our components
+- [Storybook](https://storybook.js.org/) to document and manually test our components in an isolated playground
+
+For more information see our [frontend testing guide](testing.md). Use these commands to run the tests:
 
 {% tabs %}
-{% tab title="Docker" %}
+{% tab title="With Docker" %}
 
-After you have started the application following the instructions above, in another terminal run:
+After starting the application following the above guidelines open new terminal windows for each of these commands:
 
 ```bash
-$ docker-compose exec webapp yarn storybook
+# run eslint
+$ docker-compose exec webapp yarn lint
 ```
-The output should look similar to this:
 
-![Storybook output](../.gitbook/assets/storybook-output.png)
+```bash
+# run unit tests
+$ docker-compose exec webapp yarn test
+```
 
-Click on the link http://localhost:3002/ to open the browser to your interactive storybook.
+```bash
+# start storybook
+$ docker-compose exec webapp yarn storybook
+```
+
+You can then visit the Storybook playground on `http://localhost:3002`
 
 {% endtab %}
 
 {% tab title="Without Docker" %}
-Run the following command:
+
+After starting the application following the above guidelines open new terminal windows for each of these commands:
 
 ```bash
-# in webapp/
-yarn storybook
+# run eslint
+$ yarn lint
 ```
 
-Open http://localhost:3002/ in your browser
+```bash
+# run unit tests
+$ yarn test
+```
 
-{% endtab %}
-{% endtabs %}
+```bash
+# start storybook
+$ yarn storybook
+```
 
+You can then visit the Storybook playground on `http://localhost:3002`
 
+{% endtab %}
+{% endtabs %}
 
-## Styleguide
+## Styleguide Migration
 
-All reusable Components \(for example avatar\) should be done inside the [Nitro-Styleguide](https://github.com/Human-Connection/Nitro-Styleguide) repository.
+We are currently in the process of migrating our styleguide components and design tokens from the [Nitro Styleguide](https://github.com/Human-Connection/Nitro-Styleguide) into the main [Human Connection repository](https://github.com/Human-Connection/Human-Connection) and refactoring our components in the process. During this migration our new components will live in a `view` folder to separate them from the old, yet untouched components.
 
-![Styleguide Screenshot](../.gitbook/assets/screenshot-styleguide%20%281%29.png)
+### Folder Structure
 
-More information can be found here: [https://github.com/Human-Connection/Nitro-Styleguide](https://github.com/Human-Connection/Nitro-Styleguide)
+The folder structure we are aiming for is based on the [directory setup proposed by Nuxt.js](https://nuxtjs.org/guide/directory-structure):
 
-If you need to change something in the styleguide and want to see the effects on the frontend immediately, then we have you covered. You need to clone the styleguide to the parent directory `../Nitro-Styleguide` and run `yarn && yarn run dev`. After that you run `yarn run dev:styleguide` instead of `yarn run dev` and you will see your changes reflected inside the frontend!
+- **assets** contains icons, images and logos in `svg` format
+- **components** are the generic building blocks of the app – small, reusable and usually not coupled to state
+- **features** are composed of components but tied to a particular function of the app (e.g. `comment` or `post`)
+- **layouts** can use components to create layout templates for pages
+- **pages** are the entry points for all `routes` in the app and are composed of layouts, features and components
+- **styles** holds all shared SCSS files such as `variables` and `mixins`
```


I haven't seen the `features` folder so far - @alina-beck have you been inspired by some other projects or did you come up with this on your own?

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-11-04T09:30:15Z" title="Monday, November 4th 2019, 10:30:15 am +01:00">Nov 4, 2019</time>_

---


I know the `features` folder from some React projects – although they often include more than just components ([here, for example](https://ryanlanciaux.com/blog/2017/08/20/a-feature-based-approach-to-react-development/)).

I guess it's just a matter of naming things. (_Eaaasy!_ :grin:) Some call it `containers` and `components`, others `components` and `atoms`, others `shared` and `modules`... I picked `features` and `components` because it seemed more descriptive to me. What are your preferences @mattwr18 and @roschaefer?

The idea is, as @mattwr18 already pointed out, to separate `stateful` from `stateless` components, or rather: to separate 'dumb' generic components from the ones knowing which data they are handling.